### PR TITLE
base: move to its own module

### PR DESCRIPTION
**Pull request by msgilligan**
Hopefully, this is the end of our long journey!

### CODE REVIEW

**Comment by msgilligan** [comment]
The following 4 PRs allow us to move LegacyAddressTest and SegwitAddressTest as well:

* PR #3472 
* PR #3473 
* PR #3474 
* PR #3475

When they or equivalent changes are merged to `master` we can rebase this PR.

**Comment by msgilligan** [comment]
Rebased (into 2 commits) and force-pushed.

File: base/src/main/java/org/bitcoinj/base/package-info.java
Comment (schildbach) [inline base/src/main/java/org/bitcoinj/base/package-info.java:1]: I think this needs editing in either of the two commits. After the PR, the "split into a separate JAR/module" has been done.

Comment (msgilligan) [reply base/src/main/java/org/bitcoinj/base/package-info.java:1]: Fixed and force-pushed.

**Comment by schildbach** [comment]
Merged.

**Pull request closed without merging**